xapian-core/queryparser/cjk-tokenizer.h

```diff
@@ -36,6 +36,11 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wold-style-cast"
 
+// work around missing U_PLATFORM_HAS_WINUWP_API on libicu MacOSX builds.
+#ifndef U_PLATFORM_HAS_WINUWP_API
+#define U_PLATFORM_HAS_WINUWP_API 0
+#endif
```

Comment:
This seems dubious - on platforms which would define this, it seems it gets defined by `<unicode/platform.h>` but we haven't included that yet.  So this is going to lead to a macro redefinition error on platforms which define this.

I think it would be best to just disable `-Wunused` while included these headers, like already happens for `-Wold-style-cast` above (and report this to ICU if it's not already been reported).

Reply:
Fixed in a858fcb